schema/field/numeric.go

```diff
@@ -547,6 +547,18 @@ func (b *int8Builder) Validate(fn func(int8) error) *int8Builder {
 	return b
 }
 
+// UpdateDefault sets the function that is applied to set default value
+// of the field on update. For example:
+//
+//	field.Int8("int8").
+//		Default(0).
+//		UpdateDefault(0),
+//
+func (b *int8Builder) UpdateDefault(f func() interface{}) *int8Builder {
+	b.desc.UpdateDefault = f
+	return b
+}
+
```

Comment:
Hey @godcong, the return type should match the field type, right?

```suggestion
// UpdateDefault sets the function that is applied to set default value
// of the field on update. For example:
//
//	field.Int8("int8").
//		Default(0).
//		UpdateDefault(GenNumber()),
//
func (b *int8Builder) UpdateDefault(f func() int8) *int8Builder {
	b.desc.UpdateDefault = f
	return b
}
```

Also, this file is generated and you should change it by updating the template in `schema/field/internal/numeric.tmpl `. Thanks 🙏

Reply:
I look forward to such a change, because it is easier to understand.
But I found that this change will affect the original design.
I don’t know if this destruction is acceptable.
relate test code：
```
	type CustomInt int

	f1 := func() CustomInt { return 1000 }
	fd := field.Int("id").DefaultFunc(f1).GoType(CustomInt(0)).Descriptor()
	assert.NoError(t, fd.Err)

	fd = field.Int("id").DefaultFunc(f1).Descriptor()
	assert.Error(t, fd.Err, "`var _ int = f1()` should fail")

	f2 := func() int { return 1000 }
	fd = field.Int("dir").GoType(CustomInt(0)).DefaultFunc(f2).Descriptor()
	assert.Error(t, fd.Err, "`var _ CustomInt = f2()` should fail")
```
If this can be done, I will fix the relevant changes.

Reply:
Hey @godcong, I merged your changes to https://github.com/ent/ent/pull/1770 and finished the implementation. Would you mind giving it a look before I merge it? Thanks 🙏